src/vm/hints/set.rs

```diff
@@ -139,80 +39,72 @@ pub fn set_add(
-                                        .insert(
-                                            &index_addr,
-                                            &MaybeRelocatable::Int(bigint!(i / elm_size as i32)),
-                                        )
-                                        .map_err(VirtualMachineError::MemoryError)?;
-                                    return vm
-                                        .memory
-                                        .insert(
-                                            &is_elm_in_set_addr,
-                                            &MaybeRelocatable::Int(bigint!(1)),
-                                        )
-                                        .map_err(VirtualMachineError::MemoryError);
-                                }
-                            }
-                            return vm
-                                .memory
-                                .insert(&is_elm_in_set_addr, &MaybeRelocatable::Int(bigint!(0)))
-                                .map_err(VirtualMachineError::MemoryError);
-                        }
-                        None => {
-                            return Err(VirtualMachineError::NoRangeCheckBuiltin);
-                        }
-                    }
-                };
+            let _ = vm
+                .builtin_runners
+                .iter()
+                .find(|(name, _)| name.as_str() == "range_check")
+                .ok_or(VirtualMachineError::NoRangeCheckBuiltin)?
+                .1
+                .as_any()
+                .downcast_ref::<RangeCheckBuiltinRunner>()
+                .ok_or(VirtualMachineError::NoRangeCheckBuiltin)?;
+            let elm_size = if let MaybeRelocatable::Int(ref elm_size) = maybe_rel_elm_size {
+                if elm_size.is_zero() {
+                    return Err(VirtualMachineError::ValueNotPositive(elm_size.clone()));
+                }
+                elm_size
+                    .to_usize()
+                    .ok_or_else(|| VirtualMachineError::ValueNotPositive(elm_size.clone()))?
+            } else {
+                return Err(VirtualMachineError::ExpectedInteger(
+                    maybe_rel_elm_size.clone(),
+                ));
+            };
+
+            let elm = vm
+                .memory
+                .get_range(elm_ptr, elm_size)
+                .map_err(VirtualMachineError::MemoryError)?;
+
+            if set_ptr > set_end_ptr {
+                return Err(VirtualMachineError::InvalidSetRange(
+                    set_ptr.clone(),
+                    set_end_ptr.clone(),
+                ));
             }
-            Err(VirtualMachineError::NoRangeCheckBuiltin)
+
+            let set_span = set_end_ptr.sub(set_ptr, &vm.prime)?;
+            // sub method always returns a MaybeRelocatable::Int
+            let range_limit = if let MaybeRelocatable::Int(ref range_limit) = set_span {
+                range_limit
+                    .to_i32()
```

Comment:
Shouldn't this be `to_usize`? Either that or you're returning the wrong error.

Reply:
resolved: https://github.com/lambdaclass/cleopatra_cairo/pull/289/commits/6463e1761b015ebf6c0c867d02673a7aaf105dda